platform/ios/src/MGLMapView.mm

```diff
@@ -3044,10 +3044,9 @@ - (void)removeStyleClass:(NSString *)styleClass
                 continue;
             }
             MGLAnnotationContext annotationContext = _annotationContextsByAnnotationTag.at(annotationTag);
-            if (annotationContext.annotation)
-            {
-                [annotations addObject:annotationContext.annotation];
-            }
+            NSAssert(annotationContext.annotation, @"Missing annotation for tag %u.", annotationTag);
+            [annotations addObject:annotationContext.annotation];
```

Comment:
While you’re at it, please add the same assertion and guard to the macOS implementation of `-[MGLMapView visibleAnnotationsInRect:]`. We should keep the two implementations consistent. Thanks!